### ISSUE TITLE
progress on Command line

### ISSUE THREAD

**Issue by Rajkumar-Fr** [yt-dlp/yt-dlp#2707]
### Checklist

- [X] I'm reporting a feature request
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2022.02.04**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Description

I want progress bar on yt-dlp command line

**Comment by dexeonify** [comment]
Ah yes I want that big and sexy `tqdm` progress bar!

**Comment by AmrSubZero** [comment]
Yes! why not..

**Comment by pukkandan** [comment]
@Rajkumar-Fr There is already a progress line. Pls explain

**Comment by Chitranjan6G** [comment]
# ℹ️ See "progress_hooks" in help(yt_dlp.YoutubeDL)
def my_hook(d):
    if d['status'] == 'finished':
        print('Done downloading, now post-processing ...')


ydl_opts = {
    'logger': MyLogger(),
    'progress_hooks': [my_hook],
}

with yt_dlp.YoutubeDL(ydl_opts) as ydl:
    ydl.download(URLS)


I want this progress hook in Command line.
Plz explain, how can we add ??

**Comment by Rajkumar-Fr** [comment]
# ℹ️ See "progress_hooks" in help(yt_dlp.YoutubeDL)
def my_hook(d):
    if d['status'] == 'finished':
        print('Done downloading, now post-processing ...')


ydl_opts = {
    'logger': MyLogger(),
    'progress_hooks': [my_hook],
}

with yt_dlp.YoutubeDL(ydl_opts) as ydl:
    ydl.download(URLS)



I want this progress hook in Command line.
Plz explain, how can we add ??

**Comment by pukkandan** [comment]
No! What does it even mean to implement hooks in command line!?

You can control the progress output generated with `--progress-template`, and you can always process it with external scripts.
